src/api/api.types.tsx

```diff
@@ -101,3 +101,64 @@ export interface System extends CreatedModifiedMixin {
 }
 
 export interface SystemPatch extends Partial<SystemPost> {}
+
+// ------------------------------------ CATALOGUE CATEGORIES ------------------------------------
+
+export interface AllowedValuesList {
+  type: 'list';
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  values: any[];
+}
+export type AllowedValues = AllowedValuesList;
+
+export enum CatalogueCategoryPropertyTypes {
+  Number = 'number',
+  Text = 'string',
+  Boolean = 'boolean',
+}
+
+export interface CatalogueCategoryPostProperty {
+  name: string;
+  type: CatalogueCategoryPropertyTypes;
+  unit_id?: string | null;
+  mandatory: boolean;
+  allowed_values?: AllowedValues | null;
+}
+
+export interface CatalogueCategoryPropertyPost
+  extends CatalogueCategoryPostProperty {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  default_value: any;
+}
+
+export interface CatalogueCategoryPropertyPatch {
+  name?: string;
```

Comment:
```suggestion
  name?: string | null;
```
Then again I assume this is because null is used elsewhere to indicate setting it to null, whereas null here would be the same as undefined?

Reply:
This is not consistent with what was done in the previous prs

Reply:
yeah null is the same as undefined in this case